lib/internal/modules/esm/get_format.js

```diff
@@ -112,17 +111,16 @@ function getFileProtocolModuleFormat(url, context, ignoreErrors) {
     if (defaultType === 'commonjs') { // Legacy behavior
       if (packageType === 'none' || packageType === 'commonjs') {
         return 'commonjs';
-      }
-      // If package type is `module`, fall through to the error case below
-    } else { // Else defaultType === 'module'
-      if (underNodeModules(url)) { // Exception for package scopes under `node_modules`
-        return 'commonjs';
-      }
-      if (packageType === 'none' || packageType === 'module') {
-        return getFormatOfExtensionlessFile(url);
-      } // Else packageType === 'commonjs'
-      return 'commonjs';
+      } // Else packageType === 'module'
+      return getFormatOfExtensionlessFile(url);
+    } // Else defaultType === 'module'
+    if (underNodeModules(url)) { // Exception for package scopes under `node_modules`
+      return packageType === 'module' ? 'module' : 'commonjs';
```

Comment:
```suggestion
      return packageType === 'module' ? getFormatOfExtensionlessFile(url) : 'commonjs';
```

Reply:
I just made this change in the commit I just pushed, please review.